### PR TITLE
SEO-267 Replace the category pagination with the standard one

### PR DESCRIPTION
**Pull request by gabrys**
The extension replaces the next-200, previous-200 links with the
standard pager. The new mode is not enabled if there's a "from" param
present in the URL.

### CODE REVIEW

File: extensions/wikia/CategoryPagination/CategoryPaginationViewer.class.php [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+		if ( $row ) {
+			$title = Title::newFromRow( $row );
+			$humanSortKey = $title->getCategorySortkey( $row->cl_sortkey_prefix );
+			return $humanSortKey;
+		}
+
+		return null;
+	}
+
+	public function getSectionPagingLinks( $type ) {
```
Comment (saipetch) [inline extensions/wikia/CategoryPagination/CategoryPaginationViewer.class.php]: adding position param ?

File: extensions/wikia/CategoryPagination/CategoryPaginationHooks.class.php [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+	 * @static
+	 * @param Title $title
+	 * @param Article $article
+	 * @return bool
+	 */
+	static public function onArticleFromTitle( &$title, &$article ) {
+		$app = F::app();
+
+		// Only do anything with category pages on Oasis
+		if ( !$app->checkSkin( 'oasis' ) || $title->getNamespace() != NS_CATEGORY ) {
```
Comment (saipetch) [inline extensions/wikia/CategoryPagination/CategoryPaginationHooks.class.php]: adding || empty($title) ?

File: extensions/wikia/CategoryPagination/CategoryPaginationViewer.class.php [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+				'OFFSET' => ( $page - 1 ) * $this->limit,
+				'ORDER BY' => 'cl_sortkey',
+			],
+			[ 'categorylinks' => [ 'INNER JOIN', 'cl_from = page_id' ] ]
+		);
+		$row = $res->fetchObject();
+
+		if ( $row ) {
+			$title = Title::newFromRow( $row );
+			$humanSortKey = $title->getCategorySortkey( $row->cl_sortkey_prefix );
```
Comment (saipetch) [inline extensions/wikia/CategoryPagination/CategoryPaginationViewer.class.php:88]: double check if the title is not empty ?

File: includes/CategoryViewer.php [hunk 4]
```diff
@@ -449,27 +449,16 @@ function getOtherSection() {
...
 	/**
 	 * Get the paging links for a section (subcats/pages/files), to go at the top and bottom
 	 * of the output.
 	 *
 	 * @param $type String: 'page', 'subcat', or 'file'
 	 * @return String: HTML output, possibly empty if there are no other pages
 	 */
-	private function getSectionPagingLinks( $type ) {
+	/* Wikia change: private -> public */
+	public function getSectionPagingLinks( $type ) {
```
Comment (saipetch) [inline includes/CategoryViewer.php]: using hook?

File: extensions/wikia/CategoryPagination/CategoryPaginationViewer.class.php (hunk 3 above)
Comment (gabrys) [inline extensions/wikia/CategoryPagination/CategoryPaginationViewer.class.php:88]: It's always an object.

**Comment by saipetch** [comment]
LGTM

**Pull request merged**